stable_plugins/classical_ml/data_preparation/aggregators.py

```diff
@@ -276,7 +297,55 @@ def calculation_task(self, db_id: int) -> str:
         # removes .json from file name to get the name of the attribute
         attr_name = file_name[:-5]
 
-        attribute_distances[attr_name] = json.load(file)
+        loaded_distances = json.load(file)
+
+        if missing_data_handling == MissingDataHandling.ignore:
+            # removes elements with None distance
+            loaded_distances = [
+                dist for dist in loaded_distances if dist["distance"] is not None
+            ]  # FIXME: handle all distances being None
+        elif missing_data_handling == MissingDataHandling.mean:
+            distances = [
+                dist["distance"]
+                for dist in loaded_distances
+                if dist["distance"] is not None
+            ]  # FIXME: handle all distances being None
+            mean_distance = sum(distances) / len(distances)
+
+            # replaces None distances with the mean distance
+            new_list = []
+
+            for dist in loaded_distances:
+                if dist["distance"] is None:
+                    dist["distance"] = mean_distance
+
+                new_list.append(dist)
+
+            loaded_distances = new_list
```

Comment:
Why create a new list for this? It should be fine modifying/reusing the existing one.